src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -409,3 +389,124 @@ def get_policy_assignment_completion_list(prefix, **kwargs):#pylint: disable=unu
+            resource = self.rcf.resources.get(self.resource_group_name,
+                                              self.resource_provider_namespace,
+                                              self.parent_resource_path or '',
+                                              self.resource_type,
+                                              self.resource_name,
+                                              self.api_version)
+        return resource
+
+    def delete(self):
+        if self.resource_id:
+            self.rcf.resources.delete_by_id(self.resource_id, self.api_version)
+        else:
+            self.rcf.resources.delete(self.resource_group_name,
+                                      self.resource_provider_namespace,
+                                      self.parent_resource_path or '',
+                                      self.resource_type,
+                                      self.resource_name,
+                                      self.api_version)
+
+    def tag(self, tags):
+        resource = self.get_resource()
+        # pylint: disable=no-member
+        parameters = GenericResource(
+            location=resource.location,
+            tags=tags,
+            plan=resource.plan,
+            properties=resource.properties,
+            kind=resource.kind,
+            managed_by=resource.managed_by,
+            sku=resource.sku,
+            identity=resource.identity)
+
+        if self.resource_id:
+            return self.rcf.resources.create_or_update_by_id(self.resource_id, self.api_version,
+                                                             parameters)
+        else:
+            return self.rcf.resources.create_or_update(
+                self.resource_group_name,
+                self.resource_provider_namespace,
+                self.parent_resource_path or '',
+                self.resource_type,
+                self.resource_name,
+                self.api_version,
+                parameters)
+
+    @staticmethod
+    def _resolve_api_version(rcf, resource_provider_namespace, parent_resource_path, resource_type):
+        provider = rcf.providers.get(resource_provider_namespace)
+
+        #If available, we will use parent resource's api-version
+        resource_type_str = (parent_resource_path.split('/')[0]
+                             if parent_resource_path else resource_type)
+
+        rt = [t for t in provider.resource_types
+              if t.resource_type.lower() == resource_type_str.lower()]
+        if not rt:
+            raise IncorrectUsageError('Resource type {} not found.'
+                                      .format(resource_type_str))
+        if len(rt) == 1 and rt[0].api_versions:
+            npv = [v for v in rt[0].api_versions if 'preview' not in v.lower()]
+            return npv[0] if npv else rt[0].api_versions[0]
+        else:
+            raise IncorrectUsageError(
+                'API version is required and could not be resolved for resource {}'
```

Comment:
I'm not sure if this is even possible. All resource types WILL have an api-version